Sources/Core/ObjectiveCDictionaryExtension.swift

```diff
@@ -0,0 +1,152 @@
+                ObjCIR.ifStmt("\(propIVarName) != nil") {[
+                    "[\(dictionary) setObject: ["+propIVarName + " absoluteString] forKey: @\"" + param + "\" ];"
+                    ]},
+                ObjCIR.elseStmt({[
+                    "[\(dictionary) setObject: [NSNull null] forKey: @\"" + param + "\" ];"
+                    ]
+                })
+                ]}
+        case .string(format: .some(.dateTime)):
+            return ObjCIR.scope {[
+                ObjCIR.ifStmt("\(propIVarName) != nil") {[
+                    "[\(dictionary) setObject: [[NSValueTransformer valueTransformerForName:\(dateValueTransformerKey)] transformedValue:\(propIVarName)]  forKey: @\"" + param + "\" ];"
+                    ]},
+                ObjCIR.elseStmt({[
+                    "[\(dictionary) setObject: [NSNull null] forKey: @\"" + param + "\" ];"
+                    ]
+                })
+                ]}
+        case .enumT(.integer):
+            return "[\(dictionary) setObject: [NSNumber numberWithInteger:"+propIVarName + "] forKey: @\"" + param + "\" ];"
+        case .enumT(.string):
+            return "[\(dictionary) setObject: "+enumToStringMethodName(propertyName: param, className: self.className) + "(\(propIVarName))" + " forKey: @\"" + param + "\" ];"
+        case .array(itemType: let itemType?):
+            let currentResult = "result\(counter)"
+            let currentObj = "obj\(counter)"
+            switch itemType {
+            case .reference,
+                .object:
+                return ObjCIR.scope {[
+                    "NSArray *items = \(propIVarName);",
+                    "NSMutableArray *\(currentResult) = [NSMutableArray arrayWithCapacity:items.count];",
+                    ObjCIR.forStmt("id \(currentObj) in items") { [
+                        ObjCIR.ifStmt("\(currentObj) != (id)kCFNull") { [
+                             "[\(currentResult) addObject:[ \(currentObj) dictionaryRepresentation] ];"
+                            ]}
+                        ]},
+                        "[\(dictionary) setObject:\(currentResult) forKey: @\"" + param + "\" ];"
+                    ]}
+            default:
+            return ObjCIR.scope {[
+                "NSArray *items = \(propIVarName);",
+                "NSMutableArray *\(currentResult) = [NSMutableArray arrayWithCapacity:items.count];",
+                ObjCIR.forStmt("id \(currentObj) in items") { [
+                    ObjCIR.ifStmt("\(currentObj) != (id)kCFNull") { [
+                        "[\(currentResult) addObject: \(currentObj)];"
+                        ]}
+                    ]},
+                "[\(dictionary) setObject: "+currentResult + " forKey: @\"" + param + "\" ];"
+                ]}
+            }
+        case .map(valueType: let type):
+            return ObjCIR.scope {[
+                ObjCIR.ifStmt("\(propIVarName) != nil") {[
+                            renderAddObjectStatement(param, type!, dictionary)
```

Comment:
Why do we need to force unwrap the `type` argument? We just have a separate case statement here when `itemType` on the `map` is `none`